front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -258,155 +266,260 @@ const createTrainSchedulePayload = async ({
     };
   });
 
+/**
+ * Generate properties for trainrun related caracteristics.
+ */
+const generateTrainrunProperties = async (
+  netzgrafikDto: NetzgrafikDto,
+  trainrun: TrainrunDto,
+  infraId: number,
+  dispatch: AppDispatch,
+  startTime?: string
+) => {
+  const trainrunSections = getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id);
+  const path = await generatePath(trainrunSections, netzgrafikDto.nodes, infraId, dispatch);
+  const trainrunLabels = compact(
+    uniq(
+      trainrun.labelIds.map(
+        (labelId) => netzgrafikDto.labels.find((label) => label.id === labelId)?.label
+      )
+    )
+  );
+  const startDate = calculateStartDate(
+    trainrunSections,
+    startTime ? new Date(startTime) : new Date()
+  );
+  const schedule = generateSchedule(trainrunSections, netzgrafikDto.nodes, startDate);
+
+  return { path, trainrunLabels, startDate, schedule };
+};
+
+/**
+ * Generate a TimetableItem payload for update from a TimetableItem and Trainrun properties.
+ * Handle the following cases:
+ * - if the TimetableItem is initially a PacedTrain and the frequency is still PacedTrain (`paced` duration is keep identical and step to corresponding TrainrunFrequency)
+ * - if the TimetableItem is initially a PacedTrain and the frequency is now changed to TrainSchedule (`paced` set to undefined)
+ * - if the TimetableItem is initially a TrainSchedule and the frequency is still TrainSchedule (`paced` set to undefined)
+ * - if the TimetableItem is initially a TrainSchedule and the frequency is now changed to PacedTrain (`paced` duration set to 2 hours and step to corresponding TrainrunFrequency)
+ */
+const generateTimetableItemForUpdate = async (
+  trainrun: TrainrunDto,
+  netzgrafikDto: NetzgrafikDto,
+  infraId: number,
+  dispatch: AppDispatch,
+  state: MacroEditorState
+): Promise<TrainScheduleBase | PacedTrain> => {
+  const timetableItemId = state.timetableItemIdByNgeId.get(trainrun.id)!;
+  const timetableItem = await fetchTimetableItem(timetableItemId, dispatch);
+  const { path, trainrunLabels, startDate, schedule } = await generateTrainrunProperties(
+    netzgrafikDto,
+    trainrun,
+    infraId,
+    dispatch,
+    timetableItem.start_time
+  );
+  const commonTimetableItemPayload = {
+    train_name: trainrun.name,
+    labels: trainrunLabels,
+    path,
+    start_time: startDate.toISOString(),
+    schedule,
+  };
+  const newStep = new Duration({
+    minutes: getFrequencyFromFrequencyId(state, trainrun.frequencyId).frequency,
+  }).toISOString();
+
+  let newPacedProps: PacedTrain['paced'] | undefined;
+  if (isPacedTrainResponseWithPacedTrainId(timetableItem)) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { id, timetable_id, paced, ...rest } = timetableItem;
```

Comment:
nit: You have 2 options to avoid the eslint warning (same for the one l.347)
```suggestion
    const { id: _id, timetable_id: _timetable_id, paced: _paced, ...rest } = timetableItem;
```
```suggestion
    const simplifiedTimetableItem = omit(timetableItem, ['id', 'timetable_id', 'paced']);
```

Reply:
I like the second option, thanks. Done in "front: use omit instead of eslint warning disablers"